wezterm-gui/src/termwindow/mouseevent.rs

```diff
@@ -837,7 +838,17 @@ impl super::TermWindow {
             // When hovering over a hyperlink, show an appropriate
             // mouse cursor to give the cue that it is clickable
             MouseCursor::Hand
-        } else if pane.is_mouse_grabbed() || outside_window {
+        } else if outside_window {
+            MouseCursor::Arrow
+        } else if let Some(shape) = pane.get_mouse_cursor_shape() {
+            if pane.is_mouse_grabbed() {
+                parse_mouse_cursor_shape(shape.as_str())
+            } else {
+                // If the mouse is no longer grabbed by a TUI, reset the cursor shape.
+                pane.clear_mouse_cursor_shape();
+                MouseCursor::Arrow
+            }
+        } else if pane.is_mouse_grabbed() {
```

Comment:
As it turns out, this fixed the issue of the mouse cursor shape being wrong. I have absolutely no idea if this is the right way to approach it though.

But basically; if the mouse is grabbed by whichever application is running, we defer to whatever mouse shape is set by any OSC 22 escape sequences.

If we find that a shape has been set by an OSC 22 escape sequence _but the mouse isn't grabbed_ we clear it and just use the default pointer instead.

Some thoughts:

1. I'm realizing that using `mouse_cursor_shape` for the names of this might be confusing? Should we specify that this is a shape set through an OSC 22 escape sequence somehow?
    * It's more confusing in the GUI layer, less so in the terminal state itself.
    * But it may be useful to use something like `osc_22_mouse_cursor_shape` instead of just `mouse_cursor_shape`?
2. I'm not sure this is the best place to clear the mouse cursor shape. Maybe there's an event sent when the mouse is grabbed an released that would be more appropriate?
3. As stated earlier; I have no clue if this is the right way to manage OSC 22 control sequences, but at least it behaves the way I expect.

| Before this change | After this change |
|:---:|:---:|
| <video src="https://github.com/user-attachments/assets/690078e0-4dab-4919-84f2-5018499e1b90"> | <video src="https://github.com/user-attachments/assets/95b03ab1-0c23-411d-826c-a70b1f981675"> | 

Happy to get some guidance on any improvements here!